keps/2724-topology-aware-scheduling/README.md

```diff
@@ -257,6 +264,137 @@ the "rack" topology domain (there is only one such Pod, so the requirement
 is satisfied trivially). The PodSet corresponding to the worker only
 prefers the "rack" topology domain, so it may fallback to "block".
 
+###### JobSet with ReplicatedJob instance topology
+
+To address [Story 2](#story-2) we are introducing PodSet Slices, which are
+subsets of the whole PodSet that are target to their own topology requirements.
+
+To keep solution generic (not only for JobSet) and allow to split PodSet into
+groups for other workload types, we decided to go with low-level API to allow
+user to directly specify PodSet Slice size.
+
+To achieve a result of each ReplicatedJob instance having its own topology
+domain user should align the slice size to "completions".
+
+**Example**:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: tas-example-jobset
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  replicatedJobs:
+  - name: leader
+    replicas: 1
+    template:
+      spec:
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-required-topology: cloud.provider.com/topology-block
+          spec:
+            containers:
+            - name: leader
+              image: registry.k8s.io/e2e-test-images/agnhost:2.53
+              args: ["pause"]
+  - name: workers
+    replicas: 2
+    template:
+      spec:
+        completions: 4
+        parallelism: 4
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-slice-required-topology: cloud.provider.com/topology-host
+              kueue.x-k8s.io/podset-slice-size: 4
+          spec:
+            containers:
+            - name: worker
+              image: registry.k8s.io/e2e-test-images/agnhost:2.53
+              args: ["pause"]
+```
+
+In this example there will be 8 (2 ReplicatedJob instances with 4 
+"completions" each) worker pods and we say that we want to split
+PodSet into slices with 4 pods each (so 2 slices in total) and
+each slice requires "host" topology domain.
+
+Since slice size is equal to "completions" for the Job, the end result
+is that each Job will be placed within a "host".
+
+For now PodSet Slice topology has to be required (cannot be "preferred").
```

Comment:
What do you mean `for now`? Does this mean `preferred` will be supported in Beta stage?

Reply:
That's not something we've planned, so I think it's just a poor choice of word that suggest it is planned. I'll change it to just state that without any assumption.

Reply:
We don't have currently any use-cases going beyond "required". I would suggest to indeed drop "For now", because it warrants questions. When we have such use cases we will need to update the KEP anyway.

Reply:
Thank you for an explanation. Could you add it to Non Goal section?
Once we decide to support it, we can remove it.

Reply:
Sure. I've added it there.